pkg/snapshot/snapshotter/snapshotter_test.go

```diff
@@ -902,6 +907,140 @@ var _ = Describe("Snapshotter", func() {
 				})
 			})
 		})
+
+		Describe("Scenarios to update full snapshot lease", func() {
+			var (
+				ssr   *Snapshotter
+				lease *v1.Lease
+			)
+			BeforeEach(func() {
+				healthConfig.SnapshotLeaseRenewalEnabled = true
+				healthConfig.FullSnapshotLeaseName = brtypes.DefaultFullSnapshotLeaseName
+
+				snapstoreConfig = &brtypes.SnapstoreConfig{Container: path.Join(outputDir, "default.bkp")}
+				store, err = snapstore.GetSnapstore(snapstoreConfig)
+				Expect(err).ShouldNot(HaveOccurred())
+
+				Expect(os.Setenv("POD_NAME", "test_pod")).To(Succeed())
+				Expect(os.Setenv("POD_NAMESPACE", "test_namespace")).To(Succeed())
+
+				lease = &v1.Lease{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "Lease",
+						APIVersion: "v1",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      brtypes.DefaultFullSnapshotLeaseName,
+						Namespace: os.Getenv("POD_NAMESPACE"),
+					},
+				}
+
+			})
+			AfterEach(func() {
+				Expect(os.Unsetenv("POD_NAME")).To(Succeed())
+				Expect(os.Unsetenv("POD_NAMESPACE")).To(Succeed())
+			})
+			Context("Without previous full snapshot", func() {
+				It("cannot update the lease", func() {
+					snapshotterConfig := &brtypes.SnapshotterConfig{
+						FullSnapshotSchedule: fmt.Sprintf("%d %d * * *", 0, 0),
+					}
+					ssr, err = NewSnapshotter(logger, snapshotterConfig, store, etcdConnectionConfig, compressionConfig, healthConfig, snapstoreConfig)
+					Expect(err).ShouldNot(HaveOccurred())
```

Comment:
Thanks for the review @renormalize.

I've certainly looked at the call hierarchy as part of debugging and found that `config.GetConfig` is the issue but what I'm unsure of is about the fact that multiple existing testcases which create the snapshotter the same way ([ref](https://github.com/anveshreddy18/etcd-backup-restore/blob/fd879743025e89517a9bcec447358da7d3e97f97/pkg/snapshot/snapshotter/snapshotter_test.go#L886-L887)) passes successfully in pipeline. So I think that provisioning of a cluster may not be the problem. Wdyt?

Reply:
All the other test cases which create a `Snapshotter` through the `NewSnapshotter()` invocation use the default values for `healthConfig` which is initialized by the `NewHealthConfig()` call.

For the test cases that have been added in this PR, the field `healthConfig.SnapshotLeaseRenewalEnabled` is set to `true` (it is `false` for all other test cases). This causes a function call to `GetKubernetesClientSetOrError()` in `NewSnapshotter()` which does not occur in any other test case.

`GetKubernetesClientSetOrError()` tries creating a client to a Kubernetes cluster, which would fail if there is no accessible cluster - which is what I observed in my testing.
Once I exported the `KUBECONFIG` of a running cluster, the test cases which previously failed immediately passed. Creating a client for a cluster is the only difference in `NewSnapshotter()` between the old test cases and the proposed ones in the PR.
So it is of my opinion that the lack of an accessible Kubernetes cluster is the problem.
Also see:
https://github.com/gardener/etcd-backup-restore/blob/edff58c666cef75119c10134ad9a91d92e01363c/pkg/miscellaneous/miscellaneous.go#L221-L224

which is to be used to create a client for unit tests, which is not in the call chain to fetch a client for the test cases that have been added.

Reply:
Thanks. I didn't notice this difference. I'll modify it to use fakeclient for this.